pkg/addon/controllers/addonconfiguration/controller.go

```diff
@@ -156,51 +164,47 @@ func (c *addonConfigurationController) buildConfigurationGraph(logger klog.Logge
 	// check each install strategy in status
 	var errs []error
 	for _, installProgression := range cma.Status.InstallProgressions {
-		clusters, err := c.getClustersByPlacement(installProgression.PlacementRef.Name, installProgression.PlacementRef.Namespace)
-		if errors.IsNotFound(err) {
-			logger.V(2).Info("Placement not found for addon", "placementNamespace", installProgression.PlacementRef.Namespace,
-				"placementName", installProgression.PlacementRef.Name, "addonName", cma.Name)
-			continue
-		}
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-
 		for _, installStrategy := range cma.Spec.InstallStrategy.Placements {
-			if installStrategy.PlacementRef == installProgression.PlacementRef {
-				graph.addPlacementNode(installStrategy, installProgression, clusters)
+			if installStrategy.PlacementRef != installProgression.PlacementRef {
+				continue
+			}
 
+			// get placement
+			placement, err := c.getPlacement(installProgression.PlacementRef.Name, installProgression.PlacementRef.Namespace)
```

Comment:
It seems to me this part should be done within adding placementNode

Reply:
fixed